### PR TITLE
Make warmup execute once

### PR DESCRIPTION
**Pull request by ankitkumarr**
This PR also removes the auth for /admin/warmup call.

Not 100% sure we want to do this. I need to confirm platform behavior. Still making the PR for time sake.

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Controllers/HostController.cs [hunk 1]
```diff
@@ -292,10 +293,20 @@ public IActionResult GetAdminToken()
         [HttpGet]
         [HttpPost]
         [Route("admin/warmup")]
-        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
         [RequiresRunningHost]
         public async Task<IActionResult> Warmup([FromServices] WebJobsScriptHostService hostService)
         {
+            // Endpoint only for Windows Elastic Premium or Linux App Service plans
+            if (!(_environment.IsLinuxAppService() || _environment.IsWindowsElasticPremium()))
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Controllers/HostController.cs]: Wondering why you have the IsLinuxAppService() check. Both my Windows and Linux Elastic Premium apps have sku "ElasticPremium". Why isn't checking for just that sku isn't sufficient?

Comment (ankitkumarr) [reply src/WebJobs.Script.WebHost/Controllers/HostController.cs]: Does Linux Elastic Premium have WEBSITE_SKU environment variable set to ElasticPremium? I did not find that information in my sample app. Perhaps, I looked it incorrectly?

Comment (mathewc) [reply src/WebJobs.Script.WebHost/Controllers/HostController.cs]: Look at my apps functiondev-mathewc-eptest and functiondev-mathewc-eptest-linux in Observer

File: src/WebJobs.Script.WebHost/Controllers/HostController.cs [hunk 2]
```diff
@@ -292,10 +293,20 @@ public IActionResult GetAdminToken()
...
+            {
+                return NotFound();
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Controllers/HostController.cs]: Should this be a 400? That's what I did in the GetScaleStatus endpoint when called incorrectly - https://github.com/Azure/azure-functions-host/blob/3ceb9efd6842a8d2a38002e390463c1f2dca1f80/src/WebJobs.Script.WebHost/Controllers/HostController.cs#L140.  We do the same in some other APIs

File: src/WebJobs.Script.WebHost/Controllers/HostController.cs [hunk 3]
```diff
@@ -292,10 +293,20 @@ public IActionResult GetAdminToken()
...
+            }
+
+            if (Interlocked.CompareExchange(ref _warmupExecuted, 1, 0) != 0)
+            {
+                return Ok();
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Controllers/HostController.cs]: Don't you have to update the tests? Probably should be failing currently with these changes due to incorrect sku?

File: src/WebJobs.Script.WebHost/Controllers/HostController.cs (hunk 2 above)
Comment (ankitkumarr) [reply src/WebJobs.Script.WebHost/Controllers/HostController.cs]: Fair. I think the idea is that this API is non-existent for other skus.
I guess we would eventually return 401s for those skus in future. So, 400 is probably a good idea for now.

File: src/WebJobs.Script.WebHost/Controllers/HostController.cs (hunk 3 above)
Comment (ankitkumarr) [reply src/WebJobs.Script.WebHost/Controllers/HostController.cs]: Yeah test did fail. Will update them. Made the PR quickly to get feedback and discuss next steps in the meanwhile.

File: src/WebJobs.Script.WebHost/Controllers/HostController.cs (hunk 1 above)
Comment (ankitkumarr) [reply src/WebJobs.Script.WebHost/Controllers/HostController.cs]: The site config will have the sku. But that information isn't available to the Host. Unless there's already a way to do that?
I thought Host relies on the environment variables.

Comment (mathewc) [reply src/WebJobs.Script.WebHost/Controllers/HostController.cs]: Rigth, and I assumed in Linux the same WEBSITE_SKU would have the ElasticPremium value like windows does. However, I just checked and on linux it the environment variable doesn't even exist - which surprised me.

**Pull request merged**